### PR TITLE
Flake8

### PR DESCRIPTION
**Pull request by rswarbrick**
This adds flake8 support to `lintpy.py` and enables it in CI.

### CODE REVIEW

**Comment by rswarbrick** [comment]
Ok, I think this should be in good shape now. Alex and Philipp, sorry for all the spam, but draft PRs don't generate CI runs.

With this patch, we run flake8 on every .py file that changes in a PR. I've overridden the flake8 line length to 100 (rather than the default 79), because it looks like all the code in util uses that, despite what we claim in our style guide. Maybe that needs addressing, but probably not in this commit.

**Comment by imphil** [comment]
The first commit "[util] Run lintpy.py on itself" is actually an artifact of https://github.com/lowRISC/opentitan/issues/77; depending on the Python and yapf version, you'll get different output. So if it is not needed for this PR, I'd remove this patch to avoid going back and forth on the formatting there.

Beyond that LGTM, thanks!

**Comment by rswarbrick** [comment]
Is there anyone else who should see this before it lands? I know that lots of the DV code is... quite a long way from being PEP8 compatible. Will this be a right pain for Sri and Udi, for example?

**Comment by rswarbrick** [comment]
> The first commit "[util] Run lintpy.py on itself" is actually an artifact of #77; depending on the Python and yapf version, you'll get different output. So if it is not needed for this PR, I'd remove this patch to avoid going back and forth on the formatting there.

Ah. That makes sense. I'll get rid of that and push up in the morning.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -38,13 +38,13 @@ jobs:
...
+      flake8 --version
       echo "PATH=$PATH"
     displayName: Display tool versions
   - bash: |
-      find util -iname '*.py' -print0 \
-        | xargs -0 -n1 $PWD/util/lintpy.py -f
-    # XXX: Python lint checks are disabled until Issue #313 is resolved
-    condition: False
+      fork_origin="$(git merge-base --fork-point origin/master)"
+      git diff --name-only "$fork_origin" "util/*.py" \
```
Comment (imphil) [inline azure-pipelines.yml]: Our Python style guide applies to all Python code, not only `util` code. 

Please enable this lint for all Python code, and ignore failures for the moment (`continueOnError: true`). Please file a follow-up issue for the DV team to get their scripts lint clean (include the command to run), and another follow-up to disable the "dry run" mode.

I wouldn't do a PR to reformat large chunks of DV Python code at the moment with so much work ongoing in this domain, but feel free to do so for other areas where Python code is used.

Comment (rswarbrick) [reply azure-pipelines.yml]: Ok, I'll make the change to the path. Understood about reformatting: my current plan is to make a "PEP8-fix" commit just before any proper patch to a file. That way, I can use flake8 linting to catch bugs in my own code, but I hopefully won't cause too much unnecessary churn. (That said, `git rebase` is usually excellent at sorting things out)

**Comment by imphil** [comment]
Merging as soon as CI finishes.

**Comment by rswarbrick** [comment]
For anyone who looks at this and goes "Aargh!", flake8 failures won't stop a PR from being merged. With the current setup, they will just cause warnings in the Azure pipeline (which sadly aren't visible in Github). I guess we can get more pernickety once more of the Python code is PEP8 compliant.

**Comment by rswarbrick** [comment]
Yes, that's right about W504. It's a pity that flake8 doesn't support more informative names in its exception lists but ho hum.

**Comment by imphil** [comment]
Filed follow-up for DV at https://github.com/lowRISC/opentitan/issues/1863

**Pull request merged**